File: cmd/space.go
package cmd

import (
	"fmt"

	"github.com/eugene-software/assembla-cli/internal"
	"github.com/spf13/cobra"
)

var spaceCmd = &cobra.Command{
	Use:   "space",
	Short: "Manage spaces",
}

var spaceListCmd = &cobra.Command{
	Use:   "list",
	Short: "List available spaces",
	RunE: func(cmd *cobra.Command, args []string) error {
		asJSON, _ := cmd.Flags().GetBool("json")

		data, err := Client.Get("/spaces", nil)
		if err != nil {
			return err
		}

		if asJSON {
			internal.PrintJSON(data)
		} else {
			internal.PrintTable(
				internal.ToSlice(data),
				[]string{"id", "wiki_name", "name", "status"},
				[]string{"ID", "Wiki Name", "Name", "Status"},
			)
		}
		return nil
	},
}

var spaceShowCmd = &cobra.Command{
	Use:   "show [space_id]",
	Short: "Show space details",
	Args:  cobra.MaximumNArgs(1),
	RunE: func(cmd *cobra.Command, args []string) error {
		asJSON, _ := cmd.Flags().GetBool("json")

		spaceID := Space
		if len(args) > 0 {
			spaceID = args[0]
		}
		if spaceID == "" {
			return fmt.Errorf("no space specified; use --space flag or provide space_id argument")
		}

		data, err := Client.Get(fmt.Sprintf("/spaces/%s", spaceID), nil)
		if err != nil {
			return err
		}

		if asJSON {
			internal.PrintJSON(data)
		} else {
			internal.PrintDetail(data, []string{
				"id", "wiki_name", "name", "status",
				"created_at", "updated_at", "description",
			})
		}
		return nil
	},
}

func init() {
	spaceListCmd.Flags().Bool("json", false, "Output as JSON")
	spaceShowCmd.Flags().Bool("json", false, "Output as JSON")

	spaceCmd.AddCommand(spaceListCmd)
	spaceCmd.AddCommand(spaceShowCmd)
}

File: assembla_cli/commands/space.py
from __future__ import annotations

import click

from assembla_cli.formatting import print_detail, print_json, print_table


@click.group()
def space():
    """Manage spaces."""


@space.command("list")
@click.option("--json", "as_json", is_flag=True, help="Output as JSON")
@click.pass_context
def space_list(ctx, as_json):
    """List all spaces you belong to."""
    client = ctx.obj["client"]
    data = client.get("/spaces")
    if as_json:
        print_json(data)
    else:
        print_table(
            data or [],
            columns=["id", "wiki_name", "name", "status"],
            headers=["ID", "Wiki Name", "Name", "Status"],
        )


@space.command("show")
@click.argument("space_id", required=False)
@click.option("--json", "as_json", is_flag=True, help="Output as JSON")
@click.pass_context
def space_show(ctx, space_id, as_json):
    """Show space details. Uses configured space if not provided."""
    client = ctx.obj["client"]
    space_id = space_id or ctx.obj["space"]
    data = client.get(f"/spaces/{space_id}")
    if as_json:
        print_json(data)
    else:
        print_detail(data, [
            "id", "wiki_name", "name", "status",
            "created_at", "updated_at", "description",
        ])
